open-api/rest-catalog-open-api.yaml

```diff
@@ -1482,6 +1490,34 @@ components:
       explode: false
       example: "vended-credentials,remote-signing"
 
+    page-token:
+      name: pageToken
+      in: query
+      description:
+        An opaque token which allows clients to make use of pagination by signaling to the service that they would
+        prefer requests to be paginated based on the number of items specified by pageSize.
+        
+        Clients that support pagination initiate the request by sending an empty pageToken e.g. GET /tables?pageToken
```

Comment:
I think we need to decide what exactly does it mean to be an empty token, and how it is used in the REST call.

I can see 3 ways:
1. `GET /tables?pageToken&pageSize=10`
2. `GET /tables?pageToken=&pageSize=10`
3. `GET /tables/pageToken=%02%03&pageSize=10`

(these are options described in https://stackoverflow.com/questions/21423113/passing-empty-string-parameter-in-url)

I personally think option 3 is the most clear and not ambiguous one, and similar to how we handle dots in namespace names. We also do not need to worry about the capability of different frameworks and languages in handling empty string or non-key-value query parameter. Any thoughts?

Reply:
If we are telling users to provide an empty token then why would pick option 3 where we are setting a value?

Reply:
My only concern with option 1 and 2 is that it seems to be something not universally supported. But I might be wrong. If we agree upon an empty value that is not an empty string, then any query parameter parser only needs to understand key=value syntax for query parameter, that's the most conservative approach to me. Curious what others think.

Reply:
cc @danielcweeks @rdblue

Reply:
also cc @nastra @amogh-jahagirdar

Reply:
We seem to be aligned after community sync with option 1 and option 2. It seems that when calling the `list` namespaces endpoint and adding pageToken in the following ways
```
GET /namespaces?pageToken
or
GET /namespaces?pageToken=
```
that this does not throw an error when making the request.  In my testing a service will interpret the value for pageToken in this case as empty string `""`